Code/Libraries/RESTLibrary/src/test/java/ie/tcd/wayfinders/restLibrary/LibraryTest.java

```diff
@@ -0,0 +1,223 @@
+package ie.tcd.wayfinders.restLibrary;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.http.StatusLine;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+
+import ie.tcd.wayfinders.restLibrary.Library;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+public class LibraryTest {
+
+    @Test
+    public void get200OkNoProperties() throws IOException
```

Comment:
Why is every method throwing IOException on failure?
Would it not be to throw custom exceptions, or simply the assertion errors with the appropriate error messages?

Reply:
HttpClient.execute throws an IOException in case of a problem or the connection was aborted